mockgcp/README.md

```diff
@@ -80,26 +82,36 @@ If you also use `E2E_GCP_TARGET=real` you can run against the real (non-mocked)
+   1. Run the following command to generate the golden request (`_http.log` file):
+      `E2E_KUBE_TARGET=envtest RUN_E2E=1 E2E_GCP_TARGET=real GOLDEN_REQUEST_CHECKS=1 WRITE_GOLDEN_OUTPUT=1 go test -test.count=1 -timeout 3600s -v ./tests/e2e -run TestAllInSeries/fixtures/[testname]`.
 
-Capture HTTP logs against mock GCP:
+   1. Ensure both `GOLDEN_OBJECT_CHECKS` and `GOLDEN_REQUEST_CHECKS` pass:
+      `E2E_KUBE_TARGET=envtest RUN_E2E=1 E2E_GCP_TARGET=real GOLDEN_OBJECT_CHECKS=1 GOLDEN_REQUEST_CHECKS=1 go test -test.count=1 -timeout 3600s -v ./tests/e2e -run TestAllInSeries/fixtures/[testname]`.
 
-We can consider getting HTTP logs against mockGCP after implementing the mockGCP service.
-For certain resources that are difficult to run against real GCP (for example, AttachedCluster requires an external cluster and EdgeContainer requires a physical machine, and etc), 
-it's acceptable to capture the golden HTTP logs from the mock service.
+      1. If the test doesn't pass, check the output log and identify diffs. Normalize the values if needed in
+         [tests/e2e/normalize.go](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/v1.120.1/tests/e2e/normalize.go#L66)
+         for GOLDEN_OBJECT_CHECKS and in
+         [tests/e2e/unified_test.go](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/v1.120.1/tests/e2e/unified_test.go#L523)
+         for GOLDEN_REQUEST_CHECKS.
 
-Example command: `E2E_KUBE_TARGET=envtest RUN_E2E=1 E2E_GCP_TARGET=mock GOLDEN_REQUEST_CHECKS=1 WRITE_GOLDEN_OUTPUT=1 go test -test.count=1 -timeout 3600s -v ./tests/e2e -run TestAllInSeries/fixtures/test-name`.
+1. Ensure the test works when `E2E_GCP_TARGET=mock`:
+   `E2E_KUBE_TARGET=envtest RUN_E2E=1 E2E_GCP_TARGET=mock GOLDEN_OBJECT_CHECKS=1 GOLDEN_REQUEST_CHECKS=1 go test -test.count=1 -timeout 3600s -v ./tests/e2e -run TestAllInSeries/fixtures/[testname]`.
 
-A new file named "_http.log" will be created in the test folder during the first time the command is executed, that's our golden HTTP log.
-We will compare new HTTP logs with the golden HTTP log as a part of the mockGCP test if GOLDEN_REQUEST_CHECKS is specified.
+   1. If the test doesn't pass, update the CRUD methods in the mock to match the behavior needed by the golden files.
 
-To prevent any problems when comparing with the golden logs, it is necessary to replace all generated values in the HTTP log with identical values.
+1. **In some edge cases**, we can consider getting HTTP logs against mockGCP after implementing the mockGCP service.
+   For certain resources that are difficult to run against real GCP (for example, AttachedCluster requires an
+   external cluster and EdgeContainer requires a physical machine, and etc), it's acceptable to capture the golden
+   HTTP logs from the mock service.
```

Comment:
Yes, sgtm.  And we might want to clarify that in the case of lingering diffs, we should check in the _mock_ output as otherwise the tests will fail!